### PR TITLE
add configurable timeout

### PR DESCRIPTION
**Pull request by tamarrow**
Set default to 5 seconds.

### CODE REVIEW

File: dcos/mesos.py [hunk 1]
```diff
@@ -133,7 +133,8 @@ def slave_file_read(self, slave_id, path, offset, length):
         url = self.slave_url(slave_id, 'files/read.json')
         params = {'path': path,
                   'length': length,
-                  'offset': offset}
+                  'offset': offset,
+                  'timeout': self._timeout}
```
Comment (jsancio) [inline dcos/mesos.py]: I think we shouldn't include this in `params`. Pass it as a regular function argument.

File: dcos/mesos.py [hunk 2]
```diff
@@ -177,7 +178,8 @@ def master_file_read(self, path, length, offset):
         url = self.master_url('files/read.json')
         params = {'path': path,
                   'length': length,
-                  'offset': offset}
+                  'offset': offset,
+                  'timeout': self._timeout}
```
Comment (jsancio) [inline dcos/mesos.py]: I think we shouldn't include this in `params`. Pass it as a regular function argument.

**Comment by jsancio** [comment]
LGTM. A few minor comments.

File: dcos/mesos.py (hunk 1 above)
Comment (tamarrow) [inline dcos/mesos.py]: done

File: dcos/mesos.py (hunk 2 above)
Comment (tamarrow) [inline dcos/mesos.py]: done

**Comment by tamarrow** [comment]
@jsancio updated

File: dcos/mesos.py [hunk 3]
```diff
@@ -399,7 +399,7 @@ def fetch(self, path, **kwargs):
         """
 
         url = urllib.parse.urljoin(self._base_url(), path)
-        return http.get(url, timeout=MESOS_TIMEOUT, **kwargs)
+        return http.get(url, timeout=self._timeout, **kwargs)
```
Comment (jsancio) [inline dcos/mesos.py]: There is currently no `self._timeout` in the `class Master`.

File: dcos/mesos.py [hunk 4]
```diff
@@ -233,7 +233,7 @@ def hosts(self, host):
         :rtype: dict(str, str)
         """
         url = self._path('v1/hosts/{}'.format(host))
-        return http.get(url, headers={}).json()
+        return http.get(url, headers={}, timeout=self._timeout).json()
```
Comment (jsancio) [inline dcos/mesos.py]: There is currently no `self._timeout` in `class MesosDNSClient`.

File: dcos/mesos.py (hunk 3 above)
Comment (tamarrow) [inline dcos/mesos.py]: fixed

File: dcos/mesos.py (hunk 4 above)
Comment (tamarrow) [inline dcos/mesos.py]: fixed

**Comment by tamarrow** [comment]
@jsancio updated

File: dcos/http.py [hunk 5]
```diff
@@ -69,6 +76,7 @@ def request(method,
     if 'headers' not in kwargs:
         kwargs['headers'] = {'Accept': 'application/json'}
 
+    timeout = kwargs.pop('timeout', _timeout())
```
Comment (jsancio) [inline dcos/http.py]: You want `timeout = timeout or _timeout()`.

**Comment by jsancio** [comment]
Please add some tests. If automated test are not possible please manually test this. Also, let's add tests for the new property. This is our first number property.

File: dcos/http.py (hunk 5 above)
Comment (tamarrow) [inline dcos/http.py]: done

**Comment by tamarrow** [comment]
added some tests!

**Comment by tamarrow** [comment]
@jsancio okay to merge?

**Comment by jsancio** [comment]
:shipit:

File: dcos/marathon.py [hunk 6]
```diff
@@ -87,8 +88,9 @@ class Client(object):
     :type marathon_url: str
     """
 
-    def __init__(self, marathon_url):
```
Comment (mgummelt) [inline dcos/marathon.py]: Why is this a part of the marathon interface?  We're always just using the default, so there's no need to customize it.

**Pull request merged**